qcodes/instrument_drivers/tektronix/Keithley_2450.py

```diff
@@ -108,27 +108,101 @@ def __init__(self, parent: 'Keithley2450', name: str, proper_function: str) -> N
+            docstring="The number of measurements to make when a measurement "
+                      "is requested."
+        )
+
+    def fetch(
+            self,
+            buffer_name: str = 'defbuffer1',
+            buffer_elements: Optional[list] = None
+    ) -> str:
+        """
+        This method requests the latest reading from a reading buffer.
+
+        Args:
+            buffer_name: buffer name, default is 'defbuffer1'
+            buffer_elements: available options are "DATE", "FORMatted",
+                "FRACtional", "READing", "RELative", "SEConds",
+                "SOURce", "SOURFORMatted", "SOURSTATus",
+                "SOURUNIT", "STATus","TIME", "TSTamp", "UNIT"
+        Returns:
+            The latest reading from the reading buffer
+        """
+        if buffer_elements is not None:
+            return self.ask(
+                f":FETCh? '{buffer_name}', {','.join(buffer_elements)}"
+            )
+        else:
+            return self.ask(f":FETCh? '{buffer_name}'")
+
+    def _measure(self) -> Union[float, str]:
         if not self.parent.output_enabled():
             raise RuntimeError("Output needs to be on for a measurement")
-        return self.ask(":MEASure?")
+        if self.buffer_elements is None:
+            return float(self.ask(":MEASure?"))
+        else:
+            return self.ask(f":MEASure? '{self.buffer_name}',"
+                            f"{','.join(self.buffer_elements)}")
 
     def _measure_sweep(self) -> np.ndarray:
 
         source = cast(Source2450, self.parent.source)
         source.sweep_start()
-        raw_data = self.ask(f":TRACe:DATA? 1, {self.parent.npts()}")
+        if self.buffer_elements is None:
+            raw_data = self.ask(f":TRACe:DATA? 1, {self.parent.npts()}")
+        else:
+            raw_data = self.ask(f":TRACe:DATA? 1, {self.parent.npts()},"
+                                f"'{self.buffer_name}',"
+                                f"{','.join(self.buffer_elements)}")
         # Clear the trace so we can be assured that a subsequent measurement
         # will not be contaminated with data from this run.
         self.clear_trace()
 
         return np.array([float(i) for i in raw_data.split(",")])
 
-    def clear_trace(self) -> None:
+    def make_buffer(
+            self, buffer_name: str, buffer_size: int, buffer_style: str = ''
+    ) -> None:
+        """
+        make an user defined data buffer
+        """
+        self.write(
+            f":TRACe:MAKE '{buffer_name}', {buffer_size}, {buffer_style}"
+        )
+
+    def delete_buffer(self, buffer_name: str) -> None:
+        """
+        delete an user defined data buffer
+        """
+        self.write(f":TRACe:DELete '{buffer_name}'")
+
+    def clear_trace(self, buffer_name: str = "defbuffer1") -> None:
         """
         Clear the data buffer
         """
-        self.write(":TRACe:CLEar")
+        self.write(f":TRACe:CLEar '{buffer_name}'")
```

Comment:
If user calls it without buffer name, the system will assume the user is using defbuffer1, same as the default value of the clear_trace() function. I fail to see how this will affect existing code.

Reply:
thanks for the explanation. that's indeed shouldn't be a problem now